src/core/solvers/ddp.cpp

```diff
@@ -336,7 +336,7 @@ void SolverDDP::computeGains(const std::size_t t) {
     }
     K_[t].topRows(nu).noalias() = Qxu_[t].leftCols(nu).transpose();
 
-    Eigen::Block<Eigen::MatrixXd> K = K_[t].topRows(nu);
+    auto K = K_[t].topRows(nu);
```

Comment:
Replacing this `auto` with the supposedly correct type (RowsBlockXpr / `Eigen::Block<MatrixXdRowMajor, Eigen::Dynamic, Eigen::internal::traits<MatrixXdRowMajor>::ColsAtCompileTime, true>`) results in a 3% performance loss across the board. Perhaps the compiler won't make optimisations if the type is explicitly specified? Did I select the wrong return type?

Reply:
if you have turbo boost active? It may impact the timings for no obvious reason.

Reply:
by the way, line 337, noalias is not useful.

Reply:
Thank you - I just removed the noalias that were unnecessary in f69676c24af5ff3171979a2bd2122a6e937fad8b

Reply:
Do you enable turbo boost on your machine.
For stable bench, you should disable it ;)